web_advanced_search_x2x/static/src/js/web_advanced_search_x2x.js

```diff
@@ -0,0 +1,366 @@
+                domains: search.domains,
+                contexts: search.contexts,
+                groupbys: search.groupbys || []
+            }).then(function(search)
+            {
+                var representation = self.searchview.query.reduce(function(memo, term)
+                {
+                    return _.str.sprintf(
+                        '%s%s(%s: %s)', memo, (memo ? ' ' : ''),
+                        term.attributes.category,
+                        _.reduce(term.get('values'), function(memo, value)
+                        {
+                            return memo + (memo ? ', ' : '') + value.label;
+                        }, ''));
+                }, '');
+                self.trigger('domain_selected', search.domain, representation);
+                self.destroy();
+            })
+         },
+        select_elements: function(ids)
+        {
+            var self = this;
+            return this.dataset.name_get(ids).then(function(name_gets)
+            {
+                var names = _.reduce(name_gets, function(memo, name_get)
+                {
+                    return memo + (memo ? ', ' : '') + name_get[1];
+                }, '');
+                self.trigger('domain_selected', [['id', 'in', ids]], names);
+            });
+        },
+    });
+
+    instance.web.SearchView.include({
+        build_search_data: function()
+        {
+            //Advanced.commit_search can only cope with propositions
+            //(=domain leaves),
+            //so we need to rebuild the domain if one of our CompoundDomains
+            //is involved
+            var result = this._super.apply(this, arguments);
+            _.each(result.domains, function(domain, index)
+            {
+                if(!_.isArray(domain))
+                {
+                    return;
+                }
+                var compound_domains = [], leaves = [];
+                _.each(domain, function(leaf)
+                {
+                    if(leaf instanceof instance.web.CompoundDomain)
+                    {
+                        compound_domains.push(leaf);
+                    }
+                    if(_.isArray(leaf))
+                    {
+                        leaves.push(leaf);
+                    }
+                });
+                if(compound_domains.length)
+                {
+                    var combined = new instance.web.CompoundDomain();
+                    _.each(compound_domains, function(domain)
+                    {
+                        combined.add(domain.eval());
+                    })
+                    _.each(leaves, function(leaf)
+                    {
+                        combined.add([leaf])
+                    });
+                    result.domains[index] = combined;
+                }
+            });
+            return result;
+        },
+    })
+
+
+    instance.web.PermanentSearchView = instance.web.SearchView.include({
```

Comment:
your want to modify the existing SearchView, so there's no need to keep the reference to the updated object (which is the same as `instance.web.SearchView`)